backend/src/main/java/ca/bc/gov/hlth/mohums/util/FilterUserByOrgId.java

```diff
@@ -0,0 +1,83 @@
+package ca.bc.gov.hlth.mohums.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.function.Predicate;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.SpelEvaluationException;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.util.CollectionUtils;
+
+public final class FilterUserByOrgId implements Predicate<Object> {
+
+    private static final int ABBREVIATED_USER_ENTITY_LENGTH = 64;
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(FilterUserByOrgId.class);
+
+    private static final Expression EXPRESSION = new SpelExpressionParser()
+                .parseExpression("#this['attributes']['org_details']");
+
+    private final String orgId;
+
+    public FilterUserByOrgId(String id) {
+        this.orgId = id;
+    }
+    
+    @Override
+    public String toString() {
+        return String.format("%s(%s)", getClass().getSimpleName(), orgId);
+    }
+    
+    @Override
+    public boolean test(Object userEntity) {
+        EvaluationContext context = new StandardEvaluationContext(userEntity);
+        Collection<?> orgDetails = Collections.emptyList();
+
+        try {
+            orgDetails = EXPRESSION.getValue(context, Collection.class);
+        } catch (SpelEvaluationException e) {
+            // This exception is thrown when the evaluated user entity
+            // does not contain the nested entries identified by EXPRESSION.
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
Nitpick.

The convention of checking `isDebugEnabled` before logging to DEBUG is a potential performance optimization, but we try to avoid "premature optimization" on our projects. In this case, the API we're developing will not see heavy use. We have a small, fixed set of administrators manually administering user permissions. If we have performance issues one day, we'll measure and optimize where needed. (My guess is that, if that comes, Keycloak and the database will be the problem, not this simple API proxy.)

Also, in this case it's unlikely to improve performance. We're using String parameterization in the log message, not String concatenation, and I don't believe it's actually executed unless DEBUG logging is enabled. https://stackoverflow.com/a/653846/201891